src/session_openssl.c

```diff
@@ -43,6 +43,73 @@
 #include <openssl/x509.h>
 #include <openssl/x509v3.h>
 
+#define SSLKEYLOGFILE_ENV "SSLKEYLOGFILE"
+/* The fp for the open SSLKEYLOGFILE, or NULL if not open */
+FILE *keylog_file_fp = NULL;
+uint32_t active_sessions = 0;
+
+FILE*
+nc_tls_keylog_open(void)
+{
+    char *keylog_file_name = getenv(SSLKEYLOGFILE_ENV);
+
+    if (keylog_file_name) {
+        active_sessions++;
+        if (!keylog_file_fp) {
+            keylog_file_fp = fopen(keylog_file_name, "a");
+            if (keylog_file_fp) {
+                if (setvbuf(keylog_file_fp, NULL, _IOLBF, 4096)) {
+                    fclose(keylog_file_fp);
+                    keylog_file_fp = NULL;
+                }
+            }
+        }
+    }
+    return keylog_file_fp;
+}
+
+void
+nc_tls_keylog_close(void)
+{
+    char *keylog_file_name = getenv(SSLKEYLOGFILE_ENV);
+    if (keylog_file_name) {
```

Comment:
Could also check the value of the `keylog_file_fp` variable, so that this variable may be redundant if I'm not mistaken.